larray/core/session.py

```diff
@@ -1393,87 +1394,85 @@ def display(k, v, is_metadata=False):
-    and variables of a model are defined (see examples below). These classes will allow users to
-    benefit from the so-called 'autocomplete' feature from development software such as PyCharm
-    (see the Notes section below) plus the main features of the :py:obj:`Session()` objects.
-
-    After creating an instance of a user defined 'session', some restrictions will be applied on it:
-
-    - **it is not possible to add or remove any parameter or variable,**
-    - **all non array variables (axes, groups, ...) cannot be modified,**
-    - **only values of array variables can be modified, not their axes.**
-
-    The reason of the first restriction is to avoid to select a deleted variable or
-    to miss an added one somewhere in the code when using the 'autocomplete' feature.
-    In other words, users can safely rely on the 'autocomplete' to write the model.
+class ArrayDef(ABCArray):
+    def __init__(self, axes):
+        if not all([isinstance(axis, (basestring, Axis)) for axis in axes]):
+            raise TypeError('ArrayDef only accepts string or Axis objects')
+        self.axes = axes
 
-    The reason of the second and third restrictions is to ensure the definition
-    of any variable or parameter to be constant throughout the whole code.
-    For example, a user don't need to remember that a new label has been added to
-    an axis of a given array somewhere earlier in the code (in a part of the model
-    written by a colleague). Therefore, these restrictions reduces the risk to deal
-    with unexpected error messages (like 'Incompatible Axes') and make it easier to
-    work in team.
-
-    Parameters
-    ----------
-    filepath: str, optional
-        Path where items have been saved. This can be either the path to a single file, a path to
-        a directory containing .csv files or a pattern representing several .csv files.
-    meta : list of pairs or dict or OrderedDict or Metadata, optional
-        Metadata (title, description, author, creation_date, ...) associated with the array.
-        Keys must be strings. Values must be of type string, int, float, date, time or datetime.
-
-    Notes
-    -----
-    The 'autocomplete' is a feature in which a software predicts the rest of a variable or function
-    name after a user typed the first letters. This feature allows users to use longer but meaningful
-    variable or function names (like 'population_be' instead of 'pbe') and to avoid creating an unwanted
-    new variable by misspelling the name of a given variable (e.g. typing 'poplation = something'
-    ('population' without u) will create a new variable instead of modifying the 'population' variable).
 
+# XXX: the name of the class below is not really important since it will serve as base
+#      for the LazySession and users are expected to inherit from the LazySession when
+#      they define their own classes
+class TypedSession(Session):
```

Comment:
I already like TypedSession much better than FrozenSession. But wouldn't naming it ConstrainedSession be better (we go beyond types when we mandate precise axes).

Reply:
Good suggestion.